polaris-core/src/main/java/org/apache/polaris/core/storage/StorageLocation.java

```diff
@@ -41,16 +44,21 @@ public static StorageLocation of(String location) {
   protected StorageLocation(@Nonnull String location) {
     if (location == null) {
       this.location = null;
-    } else if (location.startsWith("file:/") || location.startsWith("/")) {
-      if (!location.startsWith(LOCAL_PATH_PREFIX)) {
-        String[] parts = location.split("/", -1);
-        // Converting "file:////abcd" through Hadoop Path and Java URI will still yield "file:///abcd"
-        parts[0] = LOCAL_PATH_PREFIX;
-        location = String.join("/", parts);
-      }
-      this.location = new Path(location).toUri().toString();
+      return;
+    }
+    String scheme = scheme(location);
+    String path;
+    if (scheme == null) {
+      path = location;
+      scheme = "file";
     } else {
-      this.location = new Path(location).toString();
```

Comment:
Looks like this branch used to cover relative paths, but now it won't

Reply:
That's correct. Trying to deal with these is only going to muddy the waters further here if we try to implement it manually :( And I don't think this is a case that we really need to care heavily for cloud storage providers given that they are really just pseudo-filesystems.

If this is a strong requirement, I heavily recommend that we go back to Hadoop's `Path` class instead of trying this manually....